Kernel/Arch/aarch64/Processor.h

```diff
@@ -186,6 +186,9 @@ class Processor {
 
 private:
     u32 m_in_critical { 0 };
+
+    // FIXME: This shouldn't be a FlatPtr
+    FlatPtr m_in_irq { 0 };
```

Comment:
Does anything ever write to this?

Reply:
Nope not yet, the current code has no way to differentiate between irqs, fiqs and synchronous exceptions (syscalls). I've added that comment to the FIXME.